Activitypub and AT Protocol Discourse statement (#48)
swicg/general: statements/2025-09-05-activitypub-and-atproto-discourse.md

# Statement on discourse about ActivityPub and AT Protocol

> Date: 6th September 2025

Reviewer: I want to start off by thanking Emelia Smith (ES below) for her hard work in writing and circulating this letter. I appreciate and support the sentiment of being collegial with other protocol developers.

> A community letter on the recent heated discourse about ActivityPub and AT Protocol, co-signed by the people working on or with the respective protocols.

Reviewer: This letter speaks from a POV as if it is the consensus position of the Social Web Community Group, so I'm going to give replies with that framing.

If the idea is to instead have an area of the SocialCG GitHub repo for non-consensus position papers and letters, I think it'd be a good idea to come up with a process about how and when to do that.

Reviewer: I also want to note that this document is kind of unique for our group. We've published resolutions, Reports and report drafts, meeting notes, source code, and a lot of individual communications like issues, comments and emails. We also have the web site activitypub.rocks . I don't think we've ever published a document like this letter; I don't know of other W3C groups that publish long-form position papers like this (but I could be mistaken!). So, I want to note that this is unexplored territory.

Reviewer 2: There is at least precedent for the TAG to comment when people are being unreasonable or unproductive (e.g. https://w3ctag.github.io/the-web-is-not-versioned/). There is no reason not to do the same here. If you have concerns, you can ask the TAG's input once you've published but there is nothing problematic whatsoever with a group doing this. In fact, it's a particularly mature and intelligent position for a group to publish.

It gives me hope.

Reviewer: @darobin oh, that's helpful, thanks. It's linked as a "Findings doc", but that links to https://www.w3.org/standards/types/#finding which doesn't have anything called "Findings". Still, interesting!

Reviewer: On our own side, we have the [Social Web Protocols](https://www.w3.org/TR/social-web-protocols/) Note. I think everyone appreciated that work when it came out.

Reviewer 3: It is not my intention to across dismissive of the statement because I do think it comes from a good place. However, I don't believe this statement from the CG is necessary because there are alternatives routes that may be worth considering (if not already):

* clarify the CG charter so that it communicates better to the outside world as to the purpose of the group and the expected incubation
* refer folks to [Vision of W3C](https://www.w3.org/TR/w3c-vision/), [W3C Conduct](https://www.w3.org/policies/code-of-conduct/), and [W3C CG Process](https://www.w3.org/community/about/process/) to remind everyone and uphold expected behaviour and shared vision

I'd also refer to [The Modern Paradigm for Standards](https://open-stand.org/about-us/principles/) first principle: Cooperation.

To be constructive, the CG can consider forming a task force (or similar) to do outreach, education, and coordinate with other devs and artists across SDOs, groups, and projects.

That said, if a statement were to be publicised between works across groups/projects, it should be stated along the lines of a "Mutual Understanding Between X and Y".

Reviewer 4: There are also the W3C Ethical Web Principles that can be referred to.
https://www.w3.org/TR/ethical-web-principles/

Reviewer 5: > It's linked as a "Findings doc", but that links to https://www.w3.org/standards/types/#finding which doesn't have anything called "Findings".

Weird, there is only one (ironic!) [instance of the word "finding"](https://www.w3.org/policies/process/#team-fo-mediation) in the w3c process doc and none in the w3c document taxonomy. but it seems plenty of w3c reports have a "findings" [section](https://www.w3.org/WAI/redesign/ut_report/findings.html) or just use the word to describe this kind of informal publication, like, the "report-out"/show-your-notes part of a Report.

In any case, I think it's fine to have informal statements or micro-Reports that contain nothing more than "findings", but i also agree with sarven and evan that if "we" === SWICG (rather than "we" === the signatories listed below, open-letter style) it would take a bit of editing to make this really a collective SWICG statement, or a summary of "findings" of a conversation within the SWICG (presumably a synthesis of email threads + github threads would be the most time-efficient way to have such a convo?)

Reviewer: > It gives me hope.

This is a really interesting response! Why does this letter give you hope, @darobin ? Could you elaborate?

Reviewer 2: It gives me hope because I know first-hand that there are many in the ActivityPub community who, when AT became available, looked into it and found that it made valuable contributions and that it put forward an architecture with very interesting properties when it comes to the democratic governance of technology. This doesn't mean that they preferred it, loved it, picked "sides" (whatever that means) — in fact many had strong critiques — but they could recognise and discuss the contribution.

Unfortunately, they found themselves drowned out by a group of what I can only charitably describe as loud idiots, gleefully relying on misinformation, whose ability to compare architectural choices is limited to how much personal brand equity they've invested into a given protocol, who couldn't discuss the relative issues with self-certifying vs origin authority models if their lives depended on it but would trumpet their opinion all the same.

Suffice it to that, for those of us who are trying to bring democracy to the internet, irrespective of which protocol(s) we prefer to work on, this has been quite unhelpful. I understand that some people really enjoy picking a banner and cheering for it — I would suggest that there's a wide offering of sports out there and most of them offer multiple teams to choose from, plenty of merch as well as support groups.

So this gives me hope because it shows the community organising to signal to the loud idiots that they might want to consider being less idiotic or, failing that, less loud. It gives me hope because we don't exactly have a surfeit of people able and willing to work on regime change on the internet. Because we have challenging problems to figure out for which we can't dismiss novel, constructive contributions. Because we're not winning — but I know that we could be, with more of the kind of attitude that @ThisIsMissEm brings to this statement.

Reviewer: @darobin Thanks.

---

> Over the past few months in particular, the [Social Web Community Group](https://www.w3.org/community/socialcg/) has seen an increase in heated discussions online that have been arguing protocol superiority and creating conflict between [ActivityPub](https://www.w3.org/TR/activitypub/) and [AT Protocol](https://atproto.com/), or trying to promote one over the other. These discussions have generally not been productive, created contention within the community that stands in the way of collaboration, and been a hotbed for conflict, disagreements, and misinformation. There has often been significant biases exhibited within these conversations.

Reviewer: -0

This language is disproportionate: "heated", "arguing", "conflict", "contention", "hotbed", "disagreements", "misinformation".

-1

It's OK to promote the work we do here at the CG. That's not a problem, and shouldn't be called out as unacceptable behaviour.

-1

This letter was published about a week after a Group member, Eugen Rochko, leader of Mastodon, had an exchange with Mike Masnick, board member at Bluesky. Among the topics was the question of how Eugen talked about the Bluesky network topology. This was reported in the tech press: https://techcrunch.com/2025/08/29/mastodon-says-it-doesnt-have-the-means-to-comply-with-age-verification-laws/

The main reason a Group at the W3C would have something to say about any kind of online discussions is if a) group members are involved and b) the discussions violate the W3C code of conduct. If that's the case, it might be better to pursue this as a code of conduct issue, rather than as an open letter. 

If this letter is not about a particular incident, we should add a sentence to the end of this paragraph that makes it clear that the letter is not a "callout" and not about disciplining any particular member of the Group.

Reviewer 3: The statement also remains vague to me, and it is unclear whether the signatures are promising to uphold expected behaviour (and be accountable, and how?) as CG participants.

If the general incidents or behaviours mentioned are happening outside of the CG, it begs the question of why the CG should issue this statement to begin with.

Reviewer 5: the language _might_ be appropriate or not, i undersigned thinking we were referring to general tendencies and microblogging drama that gets overblown and taken as representative.  if i had interpreted or known this to be about eugen/masnick, or specific incidents, i might have taken issue with the language. but i'm with sarven here, probably better to be specific about the context before judging it or diagnosing it.

> ActivtyPub in its current usage does make different design decisions to AT Protocol, but ActivityPub is not necessarily that different from AT Protocol: both are open social web protocols. 

Reviewer: ActivityPub is misspelled here.

> ActivtyPub in its current usage does make different design decisions to AT Protocol, but ActivityPub is not necessarily that different from AT Protocol: both are open social web protocols. 

Reviewer 5: ```suggestion
ActivityPub in its current usage does make different design decisions to AT Protocol, but ActivityPub is not necessarily that different from AT Protocol: both are open social web protocols. 
```

There is an entire section of the ActivityPub specification that isn't as well known or widely adopted but which, at a high level, provides fairly similar ideas to those emphasized within the AT Protocol community for separation between data, identity and applications. Recently, a taskforce within the Social Web Community Group has been established to advance what is now known as the ActivityPub API. 

> Whilst we may have our differences at present, over time those gaps will narrow, as we share a lot more in common than we have differences.

Reviewer: -1

I don't think it's true that the protocols will converge, and I don't think it's material to the point of the letter.

Reviewer 3: The technicalities on intentions to converge or build on others' work can go into the CG charter. It is as simple as saying "will seek interop with such and such on these classes of products" or "will do an analysis/diff on what can be done". 

If projects or standards share ideologies, that's great, but it doesn't imply that the end result will be compatible, interoperable, or (must) converge.

> Whilst we may have our differences at present, over time those gaps will narrow, as we share a lot more in common than we have differences.

Reviewer 5: ```suggestion
Whilst we may have our tactical and strategic differences which seem significant at present, over time we hope that commonalities of values and goals will lead to the protocols complementing more and competing less.
```

Reviewer 6: I agree with this change, it fits with the message of the next paragraph that the open social web benefits more from collaboration than competition (and this is one of its strengths!).

Reviewer: I'd prefer to see some concrete next steps for collaboration. Bryan Newbold is in the CG; maybe we should invite more ATProto developers to work here with us.

> There does not have to be a “winning” protocol. We do not build a better open social web for everyone by fighting and arguing about protocol superiority. That is _**not**_ how we achieve a better open social web. Instead, we must work together, cross-pollinate and share ideas, and participate within each other's communities with respect and mutual understanding. Arguing between us only emboldens those that seek to derail and destroy efforts to build an open social web.

Reviewer: -1

I disagree on the framing and I don't think it's pertinent to the point of the letter. Even if we are competitive, we can be courteous and collaborative, too.

I don't think the appeal to mysterious and unnamed enemies is helpful, either.

Reviewer 3: "we? .. the royal we"  

I also don't understand who "we" and "between us" refer to.  

I don't find framing like "everyone by fighting and arguing" to be appropriate.  

"those that seek to derail and destroy efforts" is an appeal to vague threats.

> The practice of collaboration outside of our own groups has a long history within the standards community, whether that is with competing companies working together on standards or protocols, or collaboration between different standards bodies like the W3C and IETF.

Reviewer: -1

One of the primary differences that people bring up between ActivityPub and ATProto is governance. 

ATProto is a proprietary protocol developed and controlled by a single company. There is no patent license policy by Bluesky for ATProto. There is work underway to standardize at the IETF; but it is barely started.

ActivityPub was developed collaboratively at an open standards organization. It can be used patent and royalty free.

This is a real difference, and it's one that is important to the organization we are a part of. As a Group of the W3C, the value of openness and collaboration is intrinsic to what we do. Our [process](https://www.w3.org/policies/process/) and [mission](https://www.w3.org/mission/) matter.

I think addressing the question of governance is necessary here. In particular, does the group feel that it's a problem to point out the purpose of the W3C and the CG? That seems unreasonable.

> There has already been cross-pollination of ideas between the people working on ActivityPub and AT Protocol. For example, AT Protocol adopted an [internet draft](https://www.ietf.org/archive/id/draft-parecki-oauth-client-id-metadata-document-03.html) that was originally written to support the ActivityPub ecosystem, and projects within ActivityPub have adopted some ideas on [content labeling](https://github.com/swicg/activitypub-trust-and-safety/issues/84) and [starter packs](https://fedidevs.com/starter-packs/) from the AT Protocol ecosystem.

Reviewer: -0

I don't think we should only be courteous to people working on projects that are technically similar to our own.

-1

If collaboration is being proposed, it would be good to include concrete next steps. Some possible steps: inviting ATProto developers to join the CG. Having informal meetings between ActivityPub and ATProto devs.

Reviewer 3: Most of this seems like an argument towards why a TF may be of interest to some folks in the CG and the kinds of tasks they may want to take on, and share that information across groups.

> Both ActivityPub and AT Protocol can and do co-exist. This co-existence is perhaps best emphasized by the outstanding work of [Bridgy Fed](https://fed.brid.gy/) project, which connects ActivityPub, AT Protocol, and other protocols together allowing for interoperability and community that crosses between protocols. If you wanted to summarise this letter on a t-shirt, it would be “People > Protocols > Platforms”.

Reviewer: -1

If this document can be summarized into a single sentence or phrase, then we should agree on that single sentence or phrase as a resolution of the group. Otherwise, I think it makes sense to drop the last sentence.

Reviewer 3: Perhaps Design Principle's [Priority of Constituencies](https://www.w3.org/TR/design-principles/#priority-of-constituencies) will come across more grounded than tshirt ideas? =)

> This statement is a call for cooling the temperature of discussions and a reminder to be respectful of each other and the huge amount of work everyone is putting in to build a better open social web. We do not win by tearing each other down, which only emboldens and empowers those who do not want either protocol to succeed.

Reviewer: -1

This guidance is vague. In particular, I think it should be clear whether we are giving guidance to CG members; to the wider community of ActivityPub developers; to ActivityPub users; or to the general public.

We have a very clear code of conduct for this group:

https://www.w3.org/policies/code-of-conduct/

I'd like to see the recommendations on behaviour stick to the code of conduct.

Bryan Newbold of Bluesky and Wes Biggs of Project Liberty are both [members](https://www.w3.org/community/socialcg/participants) of the Social Web Community Group, so our COC extends to them and their companies. I think it'd be perfectly reasonable to extend our application of the code of conduct to cover others in the Bluesky and DSNP developer community, or even other groups without member representation at the CG.

I don't think we have standing to give instructions to the rest of the ActivityPub world who aren't members of this group. We can set a good example, though.

Again, here, I don't think we need to appeal to mysterious enemies. We can have better and more professional interactions regardless.

Reviewer 3: I also find the guidance unclear, and the real work is for those who deeply care about cultivating a healthy community to put in place / do whatever is necessary.

> This statement was written following an initial discussion at this month's [Social Web Community Group meeting](https://github.com/swicg/meetings/tree/main/2025-09-05), and has been reviewed by several members of the CG.

Reviewer: -1

This does not reflect my experience of that meeting. At the meeting, ES asked to put the topic on the agenda; we did not get to this agenda item before the hour was up. When the meeting was finished, ES asked that it be prioritized for the next meeting. I left the call after this point, but no additional discussion is minuted, except to note that Emelia would be writing a draft document.

Also, we work by consensus; review by some members of the CG is not enough to get consensus.

Reviewer 3: And it is rather rushed, all things considered. There ought to be more time and reflection on what is being said and intended before publishing. A CG "statement" where presumably the majority of participants are even unaware of it is not particularly inclusive. It is in the group's best interest to take its time on anything like this (see also various W3C statements, visions, and processes, and how long it took with so many revisions and so many people involved).

> ### Co-signed by the following community members:

Reviewer: -1

We don't usually sign decisions made by the CG. If this is a position paper agreed to by consensus by the CG, it should be published without signatures.

Reviewer 3: I'd add that even if the CG were to issue a statement and include signatures, it should be limited to CG participants.

Reviewer 7: I think it is good to show the broad support across diverse protocols and groups.  If it were signed *only* by SWICG members and *not* by other groups, such as AT Proto, it might look a bit odd.  With the letter, as is, it can nicely be republished to wider audiences, assuming the license permits it.

*   Dan Appelquist (Co-Chair of the W3C Advisory Board) - [@torgo](https://mastodon.social/@torgo) / [@torgo.com](https://bsky.app/profile/torgo.com) / [https://torgo.com](https://torgo.com)
*   Kagami Sascha Rosylight (Individual Misskey contributor) — [https://fosstodon.org/@krosylight](https://fosstodon.org/@krosylight)
*   Ted Han (ATProto Community Fund & IFTAS advisor) - ([@knowtheory.net](https://bsky.app/profile/knowtheory.net) / [@ted@an.errant.cloud](https://an.errant.cloud/@ted))
*   Manton Reece (Micro.blog founder) — [@manton@manton.org](https://manton.org/activitypub/manton) / [@manton.org](https://bsky.app/profile/manton.org)
*   Sean O'Brien (Yale Privacy Lab, W3C Social CG) - @profdiggity@privacysafe.social
*   Roscoe Rubin-Rottenberg (Spark Social, ATP Airport) - [@knotbin.com](https://bsky.app/profile/knotbin.com)
*   phil (microcosm.blue) – [@bad-example.com](https://bsky.app/profile/bad-example.com)
*   Blaine Cook (New\_ Public, creator of OAuth and founding engineer of Twitter) [blaine@mastodon.social](https://mastodon.social/@blaine) / [blaine.bsky.social](http://blaine.bsky.social/)

> *   Bart-Jan Schuman (various OpenSocialWeb efforts) - @schuman.de / @[Schuman@mastodon.nl](mailto:Schuman@mastodon.nl)

Reviewer 5: ```suggestion
*   Bart-Jan Schuman (various OpenSocialWeb efforts) - @schuman.de / @[Schuman@mastodon.nl](mailto:Schuman@mastodon.nl)
*   bumblefudge (SWICG, IPFS Foundation, Decentralized Identity Foundation) - bumblefudge.com
```

> 

Reviewer 2: ```suggestion
* Robin Berjon (FreeOurFeeds, IPFS Foundation, Supramundane Agency) — [@robin.berjon.com](https://robin.berjon.com/) / [@robin@mastodon.social](https://mastodon.social/@robin)
```
